docs/available-filters/record-encryption/aws-kms/service.adoc

```diff
@@ -0,0 +1,27 @@
+:aws:  https://docs.aws.amazon.com/
+
+= AWS Key Management Service
+
+For AWS KMS, the KMS configuration looks like this.
+
+[source, yaml]
+----
+kms: AwsKmsService                                            # <1>
+kmsConfig:
+  endpointUrl: https://kms.<region>.amazonaws.com             # <2>
+  tls:                                                        # <3>
+  accessKey:
+    passwordFile: /opt/aws/accessKey                          # <4>
+  secretKey:
+    passwordFile: /opt/aws/secretKey                          # <5>
```

Comment:
I know its not really a question for this PR itself however...

How often (if ever) do we re-load the credentials? I ask as I'm wondering about support for [temporary security credentials](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html#securing_access-keys) which is particularly relevant for deployments using SAML for federated access to AWS.

Reply:
We don't currently reload credentials at all.  The get read once by the factory ( io.kroxylicious.filter.encryption.RecordEncryption#initialize).  Even if they changed on the filesystem, Kroxylicious wouldn't currently notice the change.  

I saw [temporary security credentials](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html#securing_access-keys) but I haven't really understood if/how they maps to our use-case.  I went with access key/secret key just to get the simplest thing possible to work.  If you already have insight into a better/alternative ways to use IAM, raise an issue.  Otherwise, maybe we wait for user to make a suggestion.

Reply:
No i don't have any specific suggestions just the idea they are the recommended option from AWS for service accounts. 

Let's leave it till we get some input